src/TraceEvent/Symbols/NativeSymbolModule.cs

```diff
@@ -1175,12 +1203,173 @@ public byte[] GetTypeMDTokenMap()
         private ManagedSymbolModule m_managedPdb;
         private bool m_managedPdbAttempted;
 
-        internal SymbolReader m_reader;
-        internal IDiaSession m_session;
-        private IDiaDataSource3 m_source;
-        private IDiaEnumSymbolsByAddr m_symbolsByAddr;
+        internal readonly IDiaSession m_session;
+        private readonly SymbolReader m_reader;
```

Comment:
A few style questions for this area:
1. Is it good to switch to constructors to allow for readonly fields?
2. Any reason for the bool/data field combination instead of just using Lazy<T>
3. Should the m_heapAllocationSites initializer be embedded in constructor or extracted to static method?